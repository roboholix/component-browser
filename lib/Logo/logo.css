@import url("https://fonts.googleapis.com/css?family=Open+Sans&display=swap");
.logo-container {
  background: #232323;
  color: #ccc;
  display: inline-block;
  font-family: "Open Sans";
  padding: 20px;
  position: relative;
  text-decoration: none;
  transition: text-shadow 10s, filter 10s;
  white-space: nowrap;
  user-select: none;
  z-index: 10000;
}
.logo-container {
  font-size: 16px;
}
@media screen and (min-width: 320px) {
  .logo-container {
    font-size: calc(16px + 4 * ((100vw - 320px) / 760));
  }
}
@media screen and (min-width: 1080px) {
  .logo-container {
    font-size: 20px;
  }
}

.logo-container:hover {
  cursor: pointer;
}

.logo-container--is-loading-fonts {
  visibility: hidden;
}

.logo__text--electric-blue {
  color: #eff8ff;
}

.logo__text--electric-blue:hover {
  animation: neon-fadeout 0.3s ease-in-out forwards;
}

.logo__text--electric-blue-glow {
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

.logo--eftect {
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

.logo--effect1 {
  animation: pop-in 1.1s step-end forwards;
}

.logo--effect2 {
  animation: pop-in 1.35s step-end forwards;
}

.logo--effect3 {
  animation: pop-in 1.55s step-end forwards;
}

.logo--effect4 {
  animation: pop-in 1.85s step-end forwards;
}

.logo--effect5 {
  animation: pop-in 2.25s step-end forwards;
}

.logo--effect6 {
  animation: pop-in 2.67s step-end forwards;
}

.logo--effect7 {
  animation: pop-in 2.72s step-end forwards;
}

.logo--effect8 {
  animation: pop-in 2.83s step-end forwards;
}

.logo--effect9 {
  animation: pop-in 3.1s step-end forwards;
}

.logo--effect10 {
  animation: pop-in 3.25s step-end forwards, blink 1.1s linear 2.9s 5 forwards;
}

.logo--effect:hover {
  animation: none;
  color: #000;
  filter: none;
}

.logo__glitch {
  align-items: center;
  color: #eff8ff;
  height: auto;
  justify-content: center;
  pointer-events: none;
  position: relative;
  text-align: center;
  vertical-align: middle;
  width: auto;
}

.logo__glitch:after {
  animation: noise-anim 5s linear 3.25s 2 alternate-reverse;
  background: inherit;
  clip: rect(0, 900px, 0, 0);
  color: #b9b9b9;
  content: attr(data-text);
  left: 2px;
  overflow: hidden;
  position: absolute;
  text-shadow: -1px 0 #149;
  top: 0;
}

.logo__glitch:before {
  animation: noise-anim-2 5s linear 3.25s 2 alternate-reverse;
  background: inherit;
  clip: rect(0, 900px, 0, 0);
  color: white;
  content: attr(data-text);
  left: -2px;
  overflow: hidden;
  position: absolute;
  text-shadow: 2px 0 #744499;
  top: 0;
}

.text--electric-blue-glow {
  color: #eff8ff;
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

@keyframes blink {
  0% {
    color: #000;
    filter: none;
  }
  40% {
    color: #000;
    filter: none;
  }
  41% {
    color: #fff;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
  100% {
    color: #fff;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}
@keyframes neon {
  from {
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
  to {
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}
@keyframes neon-fadeout {
  from {
    filter: drop-shadow(0 0 1px #fff) drop-shadow(0 0 3px #fff) drop-shadow(0 0 6px #fff) drop-shadow(0 0 7px #228dff88);
  }
  to {
    color: #000;
    filter: none;
  }
}
@keyframes noise-anim {
  0% {
    clip: rect(95px, 9999px, 54px, 0);
  }
  10% {
    clip: rect(49px, 9999px, 76px, 0);
  }
  20% {
    clip: rect(60px, 9999px, 27px, 0);
  }
  30% {
    clip: rect(83px, 9999px, 48px, 0);
  }
  40% {
    clip: rect(1px, 9999px, 21px, 0);
  }
  50% {
    clip: rect(78px, 9999px, 7px, 0);
  }
  60% {
    clip: rect(39px, 9999px, 43px, 0);
  }
  70% {
    clip: rect(89px, 9999px, 8px, 0);
  }
  80% {
    clip: rect(4px, 9999px, 59px, 0);
  }
  90% {
    clip: rect(57px, 9999px, 62px, 0);
  }
  100% {
    clip: rect(52px, 9999px, 17px, 0);
  }
}
@keyframes noise-anim-2 {
  0% {
    clip: rect(72px, 9999px, 62px, 0);
  }
  5% {
    clip: rect(8px, 9999px, 52px, 0);
  }
  10% {
    clip: rect(82px, 9999px, 4px, 0);
  }
  15% {
    clip: rect(59px, 9999px, 78px, 0);
  }
  20% {
    clip: rect(9px, 9999px, 5px, 0);
  }
  25% {
    clip: rect(23px, 9999px, 86px, 0);
  }
  30% {
    clip: rect(100px, 9999px, 98px, 0);
  }
  35% {
    clip: rect(37px, 9999px, 89px, 0);
  }
  40% {
    clip: rect(12px, 9999px, 77px, 0);
  }
  45% {
    clip: rect(19px, 9999px, 67px, 0);
  }
  50% {
    clip: rect(14px, 9999px, 21px, 0);
  }
  55% {
    clip: rect(34px, 9999px, 80px, 0);
  }
  60% {
    clip: rect(29px, 9999px, 26px, 0);
  }
  65% {
    clip: rect(54px, 9999px, 8px, 0);
  }
  70% {
    clip: rect(62px, 9999px, 80px, 0);
  }
  75% {
    clip: rect(27px, 9999px, 75px, 0);
  }
  80% {
    clip: rect(13px, 9999px, 48px, 0);
  }
  85% {
    clip: rect(14px, 9999px, 46px, 0);
  }
  90% {
    clip: rect(99px, 9999px, 71px, 0);
  }
  95% {
    clip: rect(17px, 9999px, 93px, 0);
  }
  100% {
    clip: rect(71px, 9999px, 33px, 0);
  }
}
@keyframes pop-in {
  from {
    opacity: 0.5;
    color: #000;
    filter: none;
  }
  to {
    opacity: 1;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}

/*# sourceMappingURL=logo.css.map */
