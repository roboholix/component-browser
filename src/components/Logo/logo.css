@import url("https://fonts.googleapis.com/css?family=Open+Sans&display=swap");
.logo-container {
  background: #232323;
  color: #ccc;
  display: inline-block;
  font-family: "Open Sans";
  padding: 20px;
  position: relative;
  text-decoration: none;
  transition: text-shadow 10s, filter 10s;
  white-space: nowrap;
  user-select: none;
  z-index: 10000;
}
.logo-container {
  font-size: 16px;
}
@media screen and (min-width: 320px) {
  .logo-container {
    font-size: calc(16px + 4 * ((100vw - 320px) / 760));
  }
}
@media screen and (min-width: 1080px) {
  .logo-container {
    font-size: 20px;
  }
}

.logo-container:hover {
  cursor: pointer;
}

.logo-container--is-loading-fonts {
  visibility: hidden;
}

.logo__text--electric-blue {
  color: #eff8ff;
}

.logo__text--electric-blue:hover {
  animation: neon-fadeout 0.3s ease-in-out forwards;
}

.logo__text--electric-blue-glow {
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

.logo--eftect {
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

.logo--effect1 {
  animation: pop-in 1.1s step-end forwards;
}

.logo--effect2 {
  animation: pop-in 1.35s step-end forwards;
}

.logo--effect3 {
  animation: pop-in 1.55s step-end forwards;
}

.logo--effect4 {
  animation: pop-in 1.85s step-end forwards;
}

.logo--effect5 {
  animation: pop-in 2.25s step-end forwards;
}

.logo--effect6 {
  animation: pop-in 2.67s step-end forwards;
}

.logo--effect7 {
  animation: pop-in 2.72s step-end forwards;
}

.logo--effect8 {
  animation: pop-in 2.83s step-end forwards;
}

.logo--effect9 {
  animation: pop-in 3.1s step-end forwards;
}

.logo--effect10 {
  animation: pop-in 3.25s step-end forwards, blink 1.1s linear 2.9s 5 forwards;
}

.logo--effect:hover {
  animation: none;
  color: #000;
  filter: none;
}

.logo__glitch {
  align-items: center;
  color: #eff8ff;
  height: auto;
  justify-content: center;
  pointer-events: none;
  position: relative;
  text-align: center;
  vertical-align: middle;
  width: auto;
}

.logo__glitch:after {
  animation: noise-anim 5s linear 3.25s 2 alternate-reverse;
  background: inherit;
  clip: rect(0, 900px, 0, 0);
  color: #b9b9b9;
  content: attr(data-text);
  left: 2px;
  overflow: hidden;
  position: absolute;
  text-shadow: -1px 0 #149;
  top: 0;
}

.logo__glitch:before {
  animation: noise-anim-2 5s linear 3.25s 2 alternate-reverse;
  background: inherit;
  clip: rect(0, 900px, 0, 0);
  color: white;
  content: attr(data-text);
  left: -2px;
  overflow: hidden;
  position: absolute;
  text-shadow: 2px 0 #744499;
  top: 0;
}

.text--electric-blue-glow {
  color: #eff8ff;
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

@keyframes blink {
  0% {
    color: #000;
    filter: none;
  }
  40% {
    color: #000;
    filter: none;
  }
  41% {
    color: #fff;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
  100% {
    color: #fff;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}
@keyframes neon {
  from {
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
  to {
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}
@keyframes neon-fadeout {
  from {
    filter: drop-shadow(0 0 1px #fff) drop-shadow(0 0 3px #fff) drop-shadow(0 0 6px #fff) drop-shadow(0 0 7px #228dff88);
  }
  to {
    color: #000;
    filter: none;
  }
}
@keyframes noise-anim {
  0% {
    clip: rect(59px, 9999px, 14px, 0);
  }
  10% {
    clip: rect(59px, 9999px, 45px, 0);
  }
  20% {
    clip: rect(67px, 9999px, 93px, 0);
  }
  30% {
    clip: rect(26px, 9999px, 32px, 0);
  }
  40% {
    clip: rect(76px, 9999px, 44px, 0);
  }
  50% {
    clip: rect(57px, 9999px, 26px, 0);
  }
  60% {
    clip: rect(37px, 9999px, 47px, 0);
  }
  70% {
    clip: rect(50px, 9999px, 67px, 0);
  }
  80% {
    clip: rect(38px, 9999px, 39px, 0);
  }
  90% {
    clip: rect(73px, 9999px, 8px, 0);
  }
  100% {
    clip: rect(85px, 9999px, 30px, 0);
  }
}
@keyframes noise-anim-2 {
  0% {
    clip: rect(93px, 9999px, 76px, 0);
  }
  5% {
    clip: rect(49px, 9999px, 31px, 0);
  }
  10% {
    clip: rect(55px, 9999px, 62px, 0);
  }
  15% {
    clip: rect(74px, 9999px, 33px, 0);
  }
  20% {
    clip: rect(69px, 9999px, 38px, 0);
  }
  25% {
    clip: rect(18px, 9999px, 84px, 0);
  }
  30% {
    clip: rect(34px, 9999px, 41px, 0);
  }
  35% {
    clip: rect(81px, 9999px, 60px, 0);
  }
  40% {
    clip: rect(100px, 9999px, 66px, 0);
  }
  45% {
    clip: rect(9px, 9999px, 7px, 0);
  }
  50% {
    clip: rect(88px, 9999px, 50px, 0);
  }
  55% {
    clip: rect(97px, 9999px, 97px, 0);
  }
  60% {
    clip: rect(44px, 9999px, 62px, 0);
  }
  65% {
    clip: rect(5px, 9999px, 9px, 0);
  }
  70% {
    clip: rect(56px, 9999px, 30px, 0);
  }
  75% {
    clip: rect(58px, 9999px, 29px, 0);
  }
  80% {
    clip: rect(35px, 9999px, 77px, 0);
  }
  85% {
    clip: rect(21px, 9999px, 14px, 0);
  }
  90% {
    clip: rect(19px, 9999px, 52px, 0);
  }
  95% {
    clip: rect(52px, 9999px, 81px, 0);
  }
  100% {
    clip: rect(98px, 9999px, 61px, 0);
  }
}
@keyframes pop-in {
  from {
    opacity: 0.5;
    color: #000;
    filter: none;
  }
  to {
    opacity: 1;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}

/*# sourceMappingURL=logo.css.map */
