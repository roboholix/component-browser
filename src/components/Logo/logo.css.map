{"version":3,"sourceRoot":"","sources":["logo.scss","../../common/styles/_fluidType.scss"],"names":[],"mappings":"AAEQ;AAWR;EAEE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AChBE;EAEE,WDDU;;ACEV;EAHF;IAII;;;AAGF;EAPF;IAQI,WDNQ;;;;AAiBhB;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;;;AAGF;EACE;IACE;IACA;;EAEF;IACE;IACA;;EAEF;IACE;IACA;;EAEF;IACE;IACA;;;AAIJ;EACE;IACI;;EAGJ;IACI;;;AAKN;EACE;IACE;;EAEF;IACE;IACA;;;AAIJ;EAGI;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;;AAKN;EAGI;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;EADF;IACE;;;AAKN;EACE;IACE;IACA;IACA;;EAEF;IACE;IACA","file":"logo.css"}