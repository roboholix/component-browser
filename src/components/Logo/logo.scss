@use '../../common/styles/fluidType' as fluidType;

@import url('https://fonts.googleapis.com/css?family=Open+Sans&display=swap');

@function strip-unit($value) {
  @return $value / ($value * 0 + 1);
}

$min_width: 320px;
$max_width: 1080px;
$min_logo_font: 16px;
$max_logo_font: 20px;

.logo-container {
  @include fluidType.fluid-font-size($min_width, $max_width, $min_logo_font, $max_logo_font);
  background: #232323;
  color: #ccc;
  display: inline-block;
  font-family: "Open Sans";
  padding: 20px;
  position: relative;
  text-decoration: none;
  transition: text-shadow 10s, filter 10s;
  white-space: nowrap;
  user-select: none;
  z-index: 10000;
}

.logo-container:hover {
  cursor: pointer;
}

.logo-container--is-loading-fonts {
  visibility: hidden;
}

.logo__text--electric-blue {
  color: #eff8ff;
}

.logo__text--electric-blue:hover {
  animation: neon-fadeout .3s ease-in-out forwards;
}

.logo__text--electric-blue-glow {
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

.logo--eftect {
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

.logo--effect1 {
  animation: pop-in 1.1s step-end forwards;
}

.logo--effect2 {
  animation: pop-in 1.35s step-end forwards;
}

.logo--effect3 {
  animation: pop-in 1.55s step-end forwards;
}

.logo--effect4 {
  animation: pop-in 1.85s step-end forwards;
}

.logo--effect5 {
  animation: pop-in 2.25s step-end forwards;
}

.logo--effect6 {
  animation: pop-in 2.67s step-end forwards;
}

.logo--effect7 {
  animation: pop-in 2.72s step-end forwards;
}

.logo--effect8 {
  animation: pop-in 2.83s step-end forwards;
}

.logo--effect9 {
  animation: pop-in 3.1s step-end forwards;
}

.logo--effect10 {
  animation: pop-in 3.25s step-end forwards, blink 1.1s linear 2.9s 5 forwards;
}

.logo--effect:hover {
  animation: none;
  color: #000;
  filter: none;
}

.logo__glitch {
  align-items: center;
  color: #eff8ff;
  height: auto;
  justify-content: center;
  pointer-events: none;
  position: relative;
  text-align: center;
  vertical-align: middle;
  width: auto;
}

.logo__glitch:after {
  animation: noise-anim 5s linear 3.25s 2 alternate-reverse;
  background: inherit;
  clip: rect(0, 900px, 0, 0);
  color: #b9b9b9;
  content: attr(data-text);
  left: 2px;
  overflow: hidden;
  position: absolute;
  text-shadow: -1px 0 #149;
  top: 0;
}

.logo__glitch:before {
  animation: noise-anim-2 5s linear 3.25s 2 alternate-reverse;
  background: inherit;
  clip: rect(0, 900px, 0, 0);
  color: white;
  content: attr(data-text);
  left: -2px;
  overflow: hidden;
  position: absolute;
  text-shadow: 2px 0 #744499;
  top: 0;
}

.text--electric-blue-glow {
  color: #eff8ff;
  filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
}

@keyframes blink {
  0% {
    color: #000;
    filter: none;
  }
  40% {
    color: #000;
    filter: none;
  }
  41% {
    color: #fff;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
  100% {
    color: #fff;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}

@keyframes neon {
  from {
      filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
    //text-shadow: 0 0 7px #fff8, 0 0 15px #fffa, 0 0 25px #228dffcc;
  }
  to {
      filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
    //text-shadow: 0 0 6px #fff8, 0 0 15px #fffa, 0 0 25px #228dffcc;
  }
}

@keyframes neon-fadeout {
  from {
    filter: drop-shadow(0 0 1px #fff) drop-shadow(0 0 3px #fff) drop-shadow(0 0 6px #fff) drop-shadow(0 0 7px #228dff88);
  }
  to {
    color: #000;
    filter: none;
  }
}

@keyframes noise-anim {
  $steps: 10;
  @for $i from 0 through $steps {
    #{percentage($i*(1/$steps))} {
      clip: rect(random(100)+px, 9999px, random(100)+px, 0);
    }
  }
}

@keyframes noise-anim-2 {
  $steps: 20;
  @for $i from 0 through $steps {
    #{percentage($i*(1/$steps))} {
      clip: rect(random(100)+px, 9999px, random(100)+px, 0);
    }
  }
}

@keyframes pop-in {
  from {
    opacity: .5;
    color: #000;
    filter: none;
  }
  to {
    opacity: 1;
    filter: drop-shadow(0 0 1px #fff8) drop-shadow(0 0 5px #fffa) drop-shadow(0 0 10px #228dffcc);
  }
}