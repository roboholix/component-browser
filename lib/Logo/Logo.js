"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _fontfaceobserver = _interopRequireDefault(require("fontfaceobserver"));

var _createClassList = _interopRequireDefault(require("../../helpers/createClassList"));

require("./logo.scss");

var _ref2 = /*#__PURE__*/_react["default"].createElement("span", null, "Component Browser");

/** A simple Logo component */
var Logo = function Logo(_ref) {
  var fontSize = _ref.fontSize,
      padding = _ref.padding;

  var _useState = (0, _react.useState)(true),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isLoadingFonts = _useState2[0],
      setIsLoadingFonts = _useState2[1];

  (0, _react.useEffect)(function () {
    var isSubscribed = true;
    var font = new _fontfaceobserver["default"]('Share Tech Mono');
    font.load().then(function () {
      if (isSubscribed) {
        setIsLoadingFonts(false);
      }
    })["catch"](function (err) {
      if (isSubscribed) {
        throw new Error(err);
      }
    });
  }, []);
  var containerClasses = (0, _createClassList["default"])([isLoadingFonts ? "logo-container--is-loading-fonts" : "", "logo-container"]);
  var customStyles = {};

  if (fontSize) {
    customStyles.fontSize = fontSize;
  }

  if (padding) {
    customStyles.padding = padding;
  }

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: containerClasses,
    style: customStyles
  }, _ref2);
};

var _default = (0, _react.memo)(Logo);

exports["default"] = _default;
Logo.displayName = 'Logo';