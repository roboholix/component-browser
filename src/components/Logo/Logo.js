import React, { memo, useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import FontFaceObserver from 'fontfaceobserver';

import createClassList from "../../helpers/createClassList";

import './logo.scss';

/** A simple Logo component */
const Logo = ({ fontSize, padding }) => {

  const [isLoadingFonts, setIsLoadingFonts] = useState(true);

  useEffect(() => {

    let isSubscribed = true;

    const font = new FontFaceObserver('Share Tech Mono');

    font
      .load()
      .then(() => {
        if(isSubscribed) { setIsLoadingFonts(false); }
      })
      .catch(err => {
        if(isSubscribed) { throw new Error(err); }
      });

  }, []);

  const containerClasses = createClassList([
    isLoadingFonts ? "logo-container--is-loading-fonts" : "",
    "logo-container"
  ]);

  const customStyles = {};

  if(fontSize) {
    customStyles.fontSize = fontSize;
  }

  if(padding) {
    customStyles.padding = padding;
  }

  return (
    <div className={containerClasses} style={customStyles}>
      <span>Component Browser</span>
    </div>
  );
};

Logo.propTypes = {
  /** A font size to be applied to the logo text */
  fontSize: PropTypes.string,
  /** Css unit(s) to apply to the padding style */
  padding: PropTypes.string
};

export default memo(Logo);

Logo.displayName = 'Logo';
