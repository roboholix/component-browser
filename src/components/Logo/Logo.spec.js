import React from "react";
import TestRenderer from 'react-test-renderer';
import { mount } from "enzyme";

import Logo from "./Logo";

describe("Logo", () => {

  test("renders without blowing up", () => {

    const component = TestRenderer.create(
      <Logo />
    ).toJSON();

    expect(component).toMatchSnapshot();

  });

  test("displays the correct text", () => {
    const wrapper = mount(<Logo />);
    expect(wrapper.text()).toContain("Component Browser");
  });


});
