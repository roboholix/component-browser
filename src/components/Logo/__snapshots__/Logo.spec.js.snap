// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders without blowing up 1`] = `
<div
  className="logo-container--is-loading-fonts logo-container"
  style={Object {}}
>
  <span>
    Component Browser
  </span>
</div>
`;
